framework/src/constraints/EqualValueEmbeddedConstraint.C

```diff
@@ -0,0 +1,369 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "EqualValueEmbeddedConstraint.h"
+#include "FEProblem.h"
+#include "DisplacedProblem.h"
+#include "AuxiliarySystem.h"
+#include "SystemBase.h"
+#include "Assembly.h"
+#include "MooseMesh.h"
+#include "Executioner.h"
+#include "AddVariableAction.h"
+
+#include "libmesh/string_to_enum.h"
+#include "libmesh/sparse_matrix.h"
+
+registerMooseObject("MooseApp", EqualValueEmbeddedConstraint);
+
+template <>
+InputParameters
+validParams<EqualValueEmbeddedConstraint>()
+{
+  MooseEnum orders(AddVariableAction::getNonlinearVariableOrders());
+  InputParameters params = validParams<NodeElemConstraint>();
+  params.addClassDescription("This is a constraint enforcing overlapping portions of two blocks to "
+                             "have the same variable value");
+  params.set<bool>("use_displaced_mesh") = false;
+  params.addParam<std::string>(
```

Comment:
This should really be a MooseEnum. Then you won't need that getFormulation() method.  Look at how _cracking_release is handled in ComputeSmearedCrackingStress.C in tensor_mechanics as a template for how to do this.